tests/test_functions.py

```diff
@@ -281,13 +283,23 @@ def it_works_with_integer_values():
 
 # TODO: Figure out how to make this test deterministic locally & on CI
 def test_array_choice():
-    df = quinn.create_df(spark,
+    # Create the DataFrame so that it can be passed to the if & else blocks
+    df = quinn.create_df(
+        spark,
         [(["a", "b", "c"], "c"), (["a", "b", "c", "d"], "a"), (["x"], "x"), ([None], None)],
         [("letters", ArrayType(StringType(), True), True), ("expected", StringType(), True)],
     )
-    actual_df = df.withColumn("random_letter", quinn.array_choice(F.col("letters"), 42))
-    # chispa.assert_column_equality(actual_df, "random_letter", "expected")
 
+    # Check if the SPARK_CONNECT_MODE_ENABLED environment variable is set and if the Spark version is less than 3.5.2.
+    # If so check for the exception and if not, run the test.
+    spark_version = spark.version
+    if spark_version < "3.5.2" and os.getenv("SPARK_CONNECT_MODE_ENABLED"):
+        with pytest.raises(Exception) as excinfo:
+            df.withColumn("random_letter", quinn.array_choice(F.col("letters"), 42))
+        assert excinfo.value.args[0] == "array_choice is not supported on Spark-Connect < 3.5.2"
+    else:
+        actual_df = df.withColumn("random_letter", quinn.array_choice(F.col("letters"), 42))
+        # chispa.assert_column_equality(actual_df, "random_letter", "expected")
```

Comment:
Why in one place we are using decorators but here we are checking an exception manually?

Reply:
Happy to use the decorator here as well. The reason I used the decorator in the `test_transformations.py` was, because more than 10+ functions leverage `array_choice` function/method and didn't want to use duplicated lines of code in those 13+ functions.

Are you okay for me to move the `check_spark_connect_compatibility` function out of the `test_transformations.py` to some other file (maybe new) and import them here in the `test_functions.py` and `test_transformations.py`?

Reply:
I think that `tests/__init__.py` is a good place

Reply:
This is done @SemyonSinchenko . Let me know what you think.